### PR TITLE
Addresses BACKLOG-40051: new endpoint execute schedule permission

### PR DESCRIPTION
**Pull request by mbrasil**
@pentaho/millenniumfalcon please review

Depends on
https://github.com/pentaho/pentaho-platform/pull/5560

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 1m 15s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl core
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 10s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl core
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/platform/web/http/api/resources/SchedulerResource.java [hunk 1]
```diff
@@ -615,6 +615,32 @@ public String doGetCanSchedule() {
...
+   * </pre>
+   *
+   * @return true or false. true indicates schedule execution is allowed and false indicates schedule execution is
+   * not allowed for the user.
+   */
+  @GET
+  @Path( "/canExecuteSchedules" )
+  @Produces( APPLICATION_JSON )
+  @StatusCodes( {
+    @ResponseCode( code = 200, condition = "Successful retrieved the scheduling permission." ),
```
Comment (afrjorge) [inline core/src/main/java/org/pentaho/platform/web/http/api/resources/SchedulerResource.java:637]: Might be better to change to `Successfully retrieved` here and for `canSchedule`

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 4m 37s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **220**, Failures: **0**, Skipped: **12** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-144/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 4m 19s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **221**, Failures: **0**, Skipped: **12** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-144/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java [hunk 2]
```diff
@@ -197,84 +198,74 @@ public Job createJob( JobScheduleRequest scheduleRequest )
   /**
    * Wrapper function around {@link SchedulerOutputPathResolver#resolveOutputFilePath()} calls
    * {@link #getSchedulerOutputPathResolver(JobScheduleRequest)} to get instance.
-   * @param scheduleRequest
```
Comment (afrjorge) [inline core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java:200]: We should complete the documentation, or leave it as is for now.

Comment (mbrasil) [reply core/src/main/java/org/pentaho/platform/web/http/api/resources/services/SchedulerService.java:200]: I understand that completing the documentation is important, however, in this particular case, I was not fully aware of the details regarding the implementation which could have resulted in inaccurate information. Leaving an empty tag in the documentation should be avoided.

**Review by Pancho7 — APPROVED** [review]

File: core/src/test/java/org/pentaho/platform/web/http/api/resources/services/SchedulerServiceTest.java [hunk 3]
```diff
@@ -380,6 +390,21 @@ public void testDoGetCanSchedule() {
...
 
+  @Test
+  public void testDoGetCanExecuteSchedule() {
+    doReturn( true ).when( schedulerService.policy ).isAllowed( SchedulerExecuteAction.NAME );
+
+    //Test 1
+    String isAllowed = schedulerService.doGetCanExecuteSchedule();
+    assertEquals( "true", isAllowed );
+
+    //Test 2
```
Comment (afrjorge) [inline core/src/test/java/org/pentaho/platform/web/http/api/resources/services/SchedulerServiceTest.java:401]: please split into two unit tests

File: core/src/test/java/org/pentaho/platform/web/http/api/resources/services/SchedulerServiceTest.java [hunk 4]
```diff
@@ -292,6 +293,15 @@ public void testTriggerNow() throws Exception {
...
 
+    IPentahoSession mockSession = mock( IPentahoSession.class );
+    doReturn( mockSession ).when( schedulerService ).getSession();
+
+    String username = "username";
+    doReturn( username ).when( job ).getUserName();
+
+    String sessionName = "notUsername";
+    doReturn( sessionName ).when( mockSession ).getName();
+
```
Comment (afrjorge) [inline core/src/test/java/org/pentaho/platform/web/http/api/resources/services/SchedulerServiceTest.java:304]: please split into two unit tests

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 54s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **223**, Failures: **0**, Skipped: **12** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-144/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by afrjorge — APPROVED** [review]

**Pull request merged**